tests/class-unittestsbootstrap.php

```diff
@@ -0,0 +1,153 @@
+		$this->set_path_props();
+		$this->set_server_props();
+
+		// load test function so tests_add_filter() is available
+		require_once $this->wp_tests_dir . '/includes/functions.php';
+
+		// load WC
+		tests_add_filter( 'muplugins_loaded', array( $this, 'load_plugins' ) );
+		tests_add_filter( 'setup_theme', array( $this, 'install_wc' ) );
+		tests_add_filter( 'option_active_plugins', [ $this, 'filter_active_plugins' ] );
+
+		// load the WP testing environment
+		require_once $this->wp_tests_dir . '/includes/bootstrap.php';
+
+		// load testing framework
+		$this->includes();
+	}
+
+
+	/**
+	 * Show errors.
+	 */
+	public function set_show_errors() {
+		ini_set( 'display_errors', 'on' );
+		error_reporting( E_ALL );
+	}
+
+	/**
+	 * Set directory paths.
+	 */
+	public function set_path_props() {
+		$this->tests_dir    = dirname( __FILE__ );
+		$this->plugin_dir   = dirname( $this->tests_dir );
+		$this->plugins_dir  = sys_get_temp_dir() . '/wordpress/wp-content/plugins';
+		$this->wp_tests_dir = sys_get_temp_dir() . '/wordpress-tests-lib';
+	}
+
+	/**
+	 * Set server props
+	 */
+	public function set_server_props() {
+		$_SERVER['REMOTE_ADDR'] = isset( $_SERVER['REMOTE_ADDR'] ) ? $_SERVER['REMOTE_ADDR'] : '';
+		$_SERVER['SERVER_NAME'] = isset( $_SERVER['SERVER_NAME'] ) ? $_SERVER['SERVER_NAME'] : 'ga_integration_test';
+	}
+
+	/**
+	 * @param array $option
+	 * @return array
+	 */
+	public function filter_active_plugins( $option ) {
+		$option[] = 'woocommerce-google-analytics-integration/woocommerce-google-analytics-integration.php';
+		return $option;
+	}
+
+	/**
+	 * Load WooCommerce
+	 */
+	public function load_plugins() {
+		require_once $this->plugins_dir . '/woocommerce/woocommerce.php';
+		require_once $this->plugin_dir . '/woocommerce-google-analytics-integration.php';
+		
+		update_option( 'woocommerce_db_version', WC()->version );
+		update_option( 'gmt_offset', -4 );
+	}
+
+	/**
+	 * Load WooCommerce for testing
+	 *
+	 * @since 2.0
+	 */
+	public function install_wc() {
+		echo 'Installing WooCommerce...' . PHP_EOL;
+
+		define( 'WP_UNINSTALL_PLUGIN', true );
+
+		include $this->plugins_dir . '/woocommerce/uninstall.php';
+
+		global $wpdb;
+		$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}woocommerce_attribute_taxonomies" );
+		$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}woocommerce_order_items" );
+		$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}woocommerce_order_itemmeta" );
+
+		\WC_Install::install();
+
+		// Ensure wc_category_lookup table exists (WC 6.5+)
+		add_action(
+			'init',
+			function() {
+				if ( class_exists( \Automattic\WooCommerce\Internal\Admin\CategoryLookup::class ) ) {
+					\Automattic\WooCommerce\Internal\Admin\CategoryLookup::instance()->regenerate();
+				}
+			},
+			11
+		);
+
+		new \WP_Roles();
+
+		WC()->init();
+
+		echo 'WooCommerce Finished Installing...' . PHP_EOL;
+
+	}
+
+	/**
+	 * Load test cases and factories
+	 *
+	 * @since 2.0
+	 */
+	public function includes() {
```

Comment:
The framework file is now being autoloaded using Composer to avoid manually including it - https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/245/commits/1a6017d19bda17270bd816d3a6a14fefcfd7613a